apps/prairielearn/src/pages/instructorCourseAdminStaff/instructorCourseAdminStaff.js

```diff
@@ -14,365 +14,343 @@ const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'
 
 const sql = sqldb.loadSqlEquiv(__filename);
 
-router.get('/', (req, res, next) => {
-  if (!res.locals.authz_data.has_course_permission_own) {
-    return next(error.make(403, 'Access denied (must be course owner)'));
-  }
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_permission_own) {
+      throw error.make(403, 'Access denied (must be course owner)');
+    }
 
-  sqldb.query(sql.select_course_users, { course_id: res.locals.course.id }, (err, result) => {
-    if (ERR(err, next)) return;
-    res.locals.course_users = result.rows;
+    const course_users = await sqldb.queryAsync(sql.select_course_users, {
+      course_id: res.locals.course.id,
+    });
+    res.locals.course_users = course_users.rows;
     res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
-  });
-});
+  }),
+);
 
-router.post('/', (req, res, next) => {
-  if (!res.locals.authz_data.has_course_permission_own) {
-    return next(error.make(403, 'Access denied (must be course owner)'));
-  }
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_permission_own) {
+      throw error.make(403, 'Access denied (must be course owner)');
+    }
 
-  if (req.body.__action === 'course_permissions_insert_by_user_uids') {
-    // Get set of unique, non-empty UIDs with no leading or trailing whitespaces
-    let uids = new Set(
-      req.body.uid
-        .split(/[\s,;]+/)
-        .map((uid) => uid.trim())
-        .filter((uid) => uid),
-    );
+    if (req.body.__action === 'course_permissions_insert_by_user_uids') {
+      // Get set of unique, non-empty UIDs with no leading or trailing whitespace.
+      let uids = new Set(
+        req.body.uid
+          .split(/[\s,;]+/)
+          .map((uid) => uid.trim())
+          .filter((uid) => uid),
+      );
 
-    // Verify there is at least one UID
-    if (uids.length === 0) return next(error.make(400, 'Empty list of UIDs'));
+      // Verify there is at least one UID
+      if (uids.size === 0) throw error.make(400, 'Empty list of UIDs');
```

Comment:
I temporarily enabled type checking during development, which revealed this bug.